internal/providers/service.go

```diff
@@ -157,6 +165,116 @@ func (p *providerService) CreateGitHubOAuthProvider(ctx context.Context, provide
 	return &provider, nil
 }
 
+// CreateGitHubAppProvider creates a GitHub App provider with an installation ID
+func (p *providerService) CreateGitHubAppProvider(
+	ctx context.Context,
+	token oauth2.Token,
+	stateData db.GetProjectIDBySessionStateRow,
+	installationID int64,
+) (*db.Provider, error) {
+	installationOwner, err := p.getInstallationOwner(ctx, installationID)
+	if err != nil {
+		return nil, fmt.Errorf("error getting installation: %w", err)
+	}
+
+	savedProvider, err := db.WithTransaction(p.store, func(qtx db.ExtendQuerier) (*db.Provider, error) {
+		// Save the installation ID and create a provider
+		savedProvider, err := qtx.CreateProvider(ctx, db.CreateProviderParams{
+			Name:       fmt.Sprintf("%s-%s", db.ProviderClassGithubApp, installationOwner.GetLogin()),
```

Comment:
To make sure I'm following this right, the name will be `github-app-stacklok` or `github-app-StacklokLabs`, for example?

Reply:
That's right